### ISSUE TITLE
[Bug] Crash after enabling telemetry via the Settings menu

### ISSUE THREAD

**Issue by Lekensteyn** [mozilla-mobile/fenix#9814]
## Steps to reproduce
1. Upgrade Firefox Beta (Fenix) to 75.0b12 from the Play Store
2. Open Settings -> Data collection, Enable "Usage and technical data". The other two options below (Marketing data and Experiments) were already enabled.

### Expected behavior
Toggle changes to enabled.

### Actual behavior
Firefox crashes.

Note that I was able to reproduce it twice, both times I hit the "Report" button in the notification, but it is not visible in `about:crashes`. After the last restart, the telemetry option is somehow enabled and toggling it makes no difference. @Rob--W was not able to reproduce it on his device either (it was already enabled prior to his upgrade).

In both cases, the crash trace from `adb -vtime -b crash` is the same (time is in CEST):
<details>
<summary>adb -vtime output</summary>

```
04-09 00:39:30.903   8998         AndroidRuntime  E  FATAL EXCEPTION: main
04-09 00:39:30.903   8998         AndroidRuntime  E  Process: org.mozilla.firefox_beta, PID: 8998
04-09 00:39:30.903   8998         AndroidRuntime  E  java.lang.IllegalStateException: New tasks should not be added because queue already started, and these newly added tasks will not execute.
04-09 00:39:30.903   8998         AndroidRuntime  E  at org.mozilla.fenix.utils.StartupTaskManager.add(StartupTaskManager.kt:5)
04-09 00:39:30.903   8998         AndroidRuntime  E  at org.mozilla.fenix.components.metrics.GleanMetricsService.start(GleanMetricsService.kt:6)
04-09 00:39:30.903   8998         AndroidRuntime  E  at org.mozilla.fenix.components.metrics.ReleaseMetricController.start(Metrics.kt:8)
04-09 00:39:30.903   8998         AndroidRuntime  E  at org.mozilla.fenix.settings.DataChoicesFragment$onCreate$1.invoke(DataChoicesFragment.kt:4)
04-09 00:39:30.903   8998         AndroidRuntime  E  at org.mozilla.fenix.settings.OnSharedPreferenceChangeListener.onSharedPreferenceChanged(OnSharedPreferenceChangeListener.kt:1)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.app.SharedPreferencesImpl$EditorImpl.notifyListeners(SharedPreferencesImpl.java:612)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.app.SharedPreferencesImpl$EditorImpl.apply(SharedPreferencesImpl.java:494)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.Preference.persistBoolean(Preference.java:8)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.TwoStatePreference.setChecked(TwoStatePreference.java:5)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.TwoStatePreference.onClick(TwoStatePreference.java:4)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.Preference.performClick(Preference.java:3)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.SwitchPreference.performClick(SwitchPreference.java:1)
04-09 00:39:30.903   8998         AndroidRuntime  E  at androidx.preference.Preference$1.onClick(Preference.java:1)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.view.View.performClick(View.java:7259)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.view.View.performClickInternal(View.java:7236)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.view.View.access$3600(View.java:801)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.view.View$PerformClick.run(View.java:27892)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.os.Handler.handleCallback(Handler.java:883)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.os.Handler.dispatchMessage(Handler.java:100)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.os.Looper.loop(Looper.java:214)
04-09 00:39:30.903   8998         AndroidRuntime  E  at android.app.ActivityThread.main(ActivityThread.java:7356)
04-09 00:39:30.903   8998         AndroidRuntime  E  at java.lang.reflect.Method.invoke(Native Method)
04-09 00:39:30.903   8998         AndroidRuntime  E  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
04-09 00:39:30.903   8998         AndroidRuntime  E  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
04-09 00:39:30.921   1308   DropBoxManagerServ..  I  add tag=data_app_crash isTagEnabled=true flags=0x2
04-09 00:39:30.923   1308    ActivityTaskManager  W  Force finishing activity org.mozilla.firefox_beta/org.mozilla.fenix.HomeActivity
04-09 00:39:30.930   1308         BroadcastQueue  W  Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.
                                                    android.gms/.stats.service.DropBoxEntryAddedReceiver
04-09 00:39:30.930   1308         BroadcastQueue  W  Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.
                                                    android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
04-09 00:39:31.007   9117                 chatty  I  uid=10126(org.mozilla.firefox) expire 1 line
04-09 00:39:31.012   1308          WindowManager  I  WIN DEATH: Window{655b8d4 u0 org.mozilla.firefox_beta/org.mozilla.fenix.HomeActivity}
04-09 00:39:31.012   837                  Zygote  I  Process 8998 exited due to signal 9 (Killed)
04-09 00:39:31.012   1308        ActivityManager  I  Process org.mozilla.firefox_beta (pid 8998) has died: fore  +50 TOP
04-09 00:39:31.015   1308        ActivityManager  W  Scheduling restart of crashed service org.mozilla.firefox_beta/org.mozilla.fenix.session.SessionNotificationService in 1000ms
<elided>
04-09 00:39:55.121   9964         AndroidRuntime  E  FATAL EXCEPTION: main
04-09 00:39:55.121   9964         AndroidRuntime  E  Process: org.mozilla.firefox_beta, PID: 9964
04-09 00:39:55.121   9964         AndroidRuntime  E  java.lang.IllegalStateException: New tasks should not be added because queue already started, and these newly added tasks will not execute.
<duplicate trace elided>
04-09 00:39:55.121   9964         AndroidRuntime  E  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
04-09 00:39:55.133   1308   DropBoxManagerServ..  I  add tag=data_app_crash isTagEnabled=true flags=0x2
04-09 00:39:55.133   1308    ActivityTaskManager  W  Force finishing activity org.mozilla.firefox_beta/org.mozilla.fenix.HomeActivity
04-09 00:39:55.144   1308         BroadcastQueue  W  Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.
                                                    android.gms/.stats.service.DropBoxEntryAddedReceiver
04-09 00:39:55.146   1308        ActivityManager  I  Showing crash dialog for package org.mozilla.firefox_beta u0
04-09 00:39:55.147   1308                 chatty  I  uid=1000(system) android.ui expire 1 line
04-09 00:39:55.147   1308         BroadcastQueue  W  Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.
                                                    android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
```
</details>

The contents of the dropbox entry does not provide much new info either:
<details>
<summary>adb shell dumpsys dropbox -p '2020-04-09 00:39:30'</summary>

```
Drop box contents: 265 entries
Max entries: 1000
Low priority rate limit period: 2000 ms
Low priority tags: {data_app_wtf, keymaster, system_server_wtf, system_app_strictmode, system_app_wtf, system_server_strictmode, data_app_strictmode, netstats}
Searching for: 2020-04-09 00:39:30

========================================
2020-04-09 00:39:30 data_app_crash (text, 2103 bytes)
Process: org.mozilla.firefox_beta
PID: 8998
UID: 10126
Flags: 0x38c83e44
Package: org.mozilla.firefox_beta v2015732379 (75.0.0-beta.4)
Foreground: Yes
Build: google/walleye/walleye:10/QQ2A.200405.005/6254899:user/release-keys

java.lang.IllegalStateException: New tasks should not be added because queue already started, and these newly added tasks will not execute.
	at org.mozilla.fenix.utils.StartupTaskManager.add(StartupTaskManager.kt:5)
	at org.mozilla.fenix.components.metrics.GleanMetricsService.start(GleanMetricsService.kt:6)
	at org.mozilla.fenix.components.metrics.ReleaseMetricController.start(Metrics.kt:8)
	at org.mozilla.fenix.settings.DataChoicesFragment$onCreate$1.invoke(DataChoicesFragment.kt:4)
	at org.mozilla.fenix.settings.OnSharedPreferenceChangeListener.onSharedPreferenceChanged(OnSharedPreferenceChangeListener.kt:1)
	at android.app.SharedPreferencesImpl$EditorImpl.notifyListeners(SharedPreferencesImpl.java:612)
	at android.app.SharedPreferencesImpl$EditorImpl.apply(SharedPreferencesImpl.java:494)
	at androidx.preference.Preference.persistBoolean(Preference.java:8)
	at androidx.preference.TwoStatePreference.setChecked(TwoStatePreference.java:5)
	at androidx.preference.TwoStatePreference.onClick(TwoStatePreference.java:4)
	at androidx.preference.Preference.performClick(Preference.java:3)
	at androidx.preference.SwitchPreference.performClick(SwitchPreference.java:1)
	at androidx.preference.Preference$1.onClick(Preference.java:1)
	at android.view.View.performClick(View.java:7259)
	at android.view.View.performClickInternal(View.java:7236)
	at android.view.View.access$3600(View.java:801)
	at android.view.View$PerformClick.run(View.java:27892)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```
</details>

### Device information

* Android device: Google Pixel 2 running Android 10 (QQ2A.200405.005)
* Fenix version: 75.0b12

**Comment by Lekensteyn** [comment]
One of the files (StartupTaskManager.kt) in the stack trace has been removed in https://github.com/mozilla-mobile/fenix/commit/be1fa8df7d5cd95270c7486987badf4438e071c7 for what is worth.

**Comment by kbrosnan** [comment]
This should be fixed by bz-1635865. If you see something similar please open a new issue with a crash id as described in https://github.com/mozilla-mobile/fenix/wiki/Logging-Crash-Information thank you for taking the time to file this.
